fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/EvaluationContextBuilder.java

```diff
@@ -89,16 +93,29 @@ public EvaluationContextBuilder setBinding(@Nonnull String name, @Nullable Objec
     /**
      * Bind an alias to a value. This mutation will be
      * reflected in the {@link EvaluationContext} returned by
-     * calling {@link #build(TypeRepository)}.
+     * calling {@link #build()}.
      *
      * @param alias the alias of the binding
      * @param value the value to associate with the name
-     * @return this <code>EvaluationContextBuilder</code>
+     * @return this {@code EvaluationContextBuilder}
      */
     public EvaluationContextBuilder setBinding(@Nonnull CorrelationIdentifier alias, @Nullable Object value) {
         return setBinding(Bindings.Internal.CORRELATION.bindingName(alias.getId()), value);
     }
 
+    /**
+     * Set the {@link TypeRepository} for this evaluation context. This is an internal method that
+     * should only be used to supply types during query execution.
+     * @param typeRepository the {@link TypeRepository} to associate with the built evaluation context
+     * @return this {@code EvaluationContextBuilder}
+     */
+    @API(API.Status.INTERNAL)
```

Comment:
I marked the new methods that deal explicitly with a `TypeRepository` as `INTERNAL`, but it also looks like this isn't used (yet?) in the new planner either, so I might be off base with whether that's correct and whether the Javadoc is correct, but at least it is something we could change in the future.